go-controller/pkg/ovn/obj_retry.go

```diff
@@ -296,6 +298,17 @@ func areResourcesEqual(objType reflect.Type, obj1, obj2 interface{}) (bool, erro
 		// For these types there is no update code, so pretend old and new
 		// objs are always equivalent and stop processing the update event.
 		return true, nil
+
+	case factory.EgressFirewallType:
+		newEgressFirewall, ok := obj1.(*egressfirewall.EgressFirewall)
+		if !ok {
+			return false, fmt.Errorf("could not cast obj1 of type interface{} to *egressfirewall.EgressFirewall")
```

Comment:
I get you are just following the previous convention here, but we should change these errors to print out the actual type of the object using %T if its not what we expected.